### PR TITLE
Stop Global Audio player from playing loose audio

### LINKED ISSUE #1537: The audio from search results keeps playing on the single result page even without the global player
## Description

The audio from search results keeps playing on the single result page even without the global player.

## Reproduction

1. On a desktop search results page (All content or Audio), click Play for one of the audio results.
2. Click on another audio result to open a single result page.
3. The audio will still be playing, and you have no way of stopping it. 

## Additional context

The audio is loaded in the Global audio player. But the player is hidden on the desktop screens.

## Resolution

- [ ] 🙋 I would be interested in resolving this bug.

### PR DESCRIPTION
**Pull request by krysal**
## Fixes

Fixes #1537 by @obulat

## Description

This PR limits the reproduction of the global audio to the 'search-audio' and its own 'audio-id' (Audio Details) routes on desktop viewports.

I added a base test for the happy path of the global audio but I don't know yet how to create the test for the actual bug, suggestions are welcome.

## Testing Instructions

Follow the steps of the linked issue and observe the audio stops.

## Checklist

- [x] My pull request has a descriptive title (not a vague title like `Update index.md`).
- [x] My pull request targets the *default* branch of the repository (`main`) or a parent feature branch.
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [x] I added or updated tests for the changes I made (if applicable).
- [x] I added or updated documentation (if applicable).
- [x] I tried running the project locally and verified that there are no visible errors.

[best_practices]:https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

File: test/playwright/e2e/global-audio.spec.ts [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+  expect(nthAudioRow).toHaveAttribute('status', 'paused')
+  return nthAudioRow
+}
+
+const play = async (audioRow: Locator) => {
+  await audioRow.locator('button[aria-label="Play"] >> visible=true').click()
+  expect(audioRow).toHaveAttribute('status', /(loading|playing)/)
+}
+
+test.describe('global audio', () => {
```
Comment (obulat) [inline test/playwright/e2e/global-audio.spec.ts:17]: I think we could also add a test case where we click play and then pause on the search page, and then check that the audio **isn't** playing on the detail page.

**Review by obulat — APPROVED** [review]
Nice, this fixes this really annoying problem!
One step closer to getting audio out of beta 🎉

File: test/playwright/e2e/global-audio.spec.ts (hunk 1 above)
Comment (krysal) [reply test/playwright/e2e/global-audio.spec.ts:17]: I had that test written but didn't like how I named it. I just added it with a better name :)

File: src/components/VGlobalAudioSection/VGlobalAudioSection.vue [hunk 2]
```diff
@@ -112,13 +113,21 @@ export default defineComponent({
...
-    /* Router observation */
-
-    const routeName = computed(() => route.value.name)
-    watch(routeName, (routeNameVal, oldRouteNameVal) => {
+    /**
+     * Router observation
+     *
+     * The player will continue only within 'search-audio' and 'audio-id' routes,
+     * and on desktop, only if the next route if the 'audio-id' page of the
+     * track currently playing, or the search page where it comes from.
```
Comment (zackkrida) [inline src/components/VGlobalAudioSection/VGlobalAudioSection.vue:121]:
```suggestion
     * and on desktop, only if the next route is the 'audio-id' page of the
     * track currently playing, or the original search result page.
```

**Review by zackkrida — APPROVED** [review]
Excellent!

**Pull request merged**